### ISSUE TITLE
Should GenAPI be removing the ToolboxItem attribute?

### ISSUE THREAD

**Issue by ahsonkhan** [dotnet/corefx#35607]
Example usages: https://github.com/dotnet/corefx/blob/77cde697f9ee9941bdc76e02b9481d781790e86a/src/System.Diagnostics.EventLog/ref/System.Diagnostics.EventLog.cs#L101-L102
https://github.com/dotnet/corefx/blob/77cde697f9ee9941bdc76e02b9481d781790e86a/src/System.Management/ref/System.Management.cs#L136-L137

Excluded here: https://github.com/dotnet/corefx/blob/77cde697f9ee9941bdc76e02b9481d781790e86a/eng/DefaultGenApiDocIds.txt#L7

Should we be emitting these or retain them?

cc @vancem

**Comment by Wraith2** [comment]
What level of information are the ref assemblies designed to provide? I had thought that they were present to provide a description of the api surface area that an ide or other metadata consuming tool could use to provide intellisense or some sort of binding. If they are for ide type use then any attribute which can affect consumer behaviour would need to be included, things like EditorBrowsable etc.

**Comment by ahsonkhan** [comment]
> If they are for ide type use then any attribute which can affect consumer behaviour would need to be included, things like EditorBrowsable etc.

I agree and understand why attributes like `EditorBrowsable` or `Obsolete` would need to be included. I am not sure about the utility of `ToolboxItem` and some of the other attributes though.

**Comment by vancem** [comment]
I know very little about ToolBoxItem, but it seems it is used in a VS designer (I vaguely remember that there was a designer for event log and system management stuff.   

I could believe the designers that use these two types are broken for .NET Core, but I really don't know.   

If I were you, if we have ToolboxItemAttribute in .NET Core then we should not try to provoke policy but uniformly simply pass attributes through in GenAPI (we don't want to be in the business of understanding what the attributes are or do, we are just making the reference version.  Thus keep all attributes you can.

For what it is worth....

**Comment by joperezr** [comment]
Just a note, I believe all of these GenAPI issues should instead be labeld as area-Infrastructure as the fix would have to go on arcade repo.

**Comment by ericstj** [comment]
@dreddy-work @Tanya-Solyanik @DustinCampbell Can you let us know if the toolbox / winforms designer depends on having ToolBoxItemAttribute on our reference assemblies?

**Comment by Tanya-Solyanik** [comment]
Most likely we will depend on having this attribute for adding items to the toolbox and for finding item bitmaps on ToolboxBitmapAttribute.  @Shyam-Gupta   - FYI
